core/src/main/java/org/elasticsearch/search/aggregations/metrics/tophits/TopHitsAggregatorFactory.java

```diff
@@ -79,6 +79,7 @@ public TopHitsAggregatorFactory(String name, Type type, int from, int size, bool
     public Aggregator createInternal(Aggregator parent, boolean collectsFromSingleBucket, List<PipelineAggregator> pipelineAggregators,
             Map<String, Object> metaData) throws IOException {
         SubSearchContext subSearchContext = new SubSearchContext(context.searchContext());
+        subSearchContext.parsedQuery(context.searchContext().parsedQuery());
```

Comment:
Actually don't worry, I saw below this that you had to change SubSearchContext and thats why this changed